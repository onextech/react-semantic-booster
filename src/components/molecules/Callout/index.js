import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Grid, Button } from 'semantic-ui-react';
import { MediaCss } from '../../../utils/responsive';
import Block from '../../atoms/Block';


const CalloutText = styled.p`
  font-size: 1.4em;
  font-weight: 300;
  line-height: 1.25;
  margin: 1.5em auto;
  opacity: 0.8;
`;

const ResponsiveRightTextColumn = styled(Grid.Column)`
  text-align: center;
  ${MediaCss.min.sm`text-align: right;`}
`;

const ResponsiveLeftTextColumn = styled(Grid.Column)`
  text-align: center;
  ${MediaCss.min.sm`text-align: left;`}
`;

const Callout = ({
  text, cta, block, button,
}) => (
  <Block vertical spacer={0.5} {...block}>
    <Grid container stackable>
      <Grid.Row verticalAlign="middle">
        <ResponsiveLeftTextColumn width={12}>
          <CalloutText>{text}</CalloutText>
        </ResponsiveLeftTextColumn>
        <ResponsiveRightTextColumn width={4}>
          <Button primary circular size="large" {...button}>{cta}</Button>
        </ResponsiveRightTextColumn>
      </Grid.Row>
    </Grid>
  </Block>
);

Callout.propTypes = {
  text: PropTypes.string.isRequired,
  cta: PropTypes.string.isRequired,
  button: PropTypes.object, // semantic-ui button props
  block: PropTypes.object, // Block component props
};

Callout.defaultProps = {
  button: undefined,
  block: undefined,
};

export default Callout;
