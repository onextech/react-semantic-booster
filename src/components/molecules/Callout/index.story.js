import React from 'react';
import { storiesOf } from '@storybook/react';
import Callout from '.';

export const text = <span>Based in Singapore. Available for event bookings <span role="img" aria-label="Victory Hand">✌🏻</span></span>;
export const cta = 'Make Booking';

storiesOf('Callout', module)
  .add('Default', () => (
    <div>
      <Callout text={text} cta={cta} />
      <Callout text={text} cta={cta} block={{ secondary: true }} />
      <Callout text={text} cta={cta} block={{ inverted: true }} />
    </div>
  ));
