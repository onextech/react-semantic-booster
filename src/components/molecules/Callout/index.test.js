import React from 'react';
import renderer from 'react-test-renderer';
import Callout from '.';
import { text, cta } from './index.story';


describe('<Callout>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Callout text={text} cta={cta} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
