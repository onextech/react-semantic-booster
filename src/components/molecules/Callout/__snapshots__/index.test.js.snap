// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout> renders correctly 1`] = `
<div
  angular={undefined}
  background={undefined}
  className="ui vertical segment sc-bdVaJa fzUeb"
  spacer={0.5}
>
  <div
    className="ui container stackable grid"
  >
    <div
      className="middle aligned row"
    >
      <div
        className="twelve wide column sc-bxivhb iiVAHN"
      >
        <p
          className="sc-bwzfXH BaQOG"
        >
          Available for bookings. Based in Singapore. Open to custom events.
        </p>
      </div>
      <div
        className="four wide column sc-htpNat uswuH"
      >
        <button
          className="ui large circular primary button"
          disabled={undefined}
          onClick={[Function]}
          role="button"
          tabIndex={undefined}
        >
          Make Booking
        </button>
      </div>
    </div>
  </div>
</div>
`;
